docs/how_to/openedx/release_upgrades.md

```diff
@@ -0,0 +1,131 @@
+included in a given deployment and which version to use in the respective environment
+stage. The `OpenLearningOpenEdxDeployment` Enum is the top level that sets the release
+name for the environment stage of a given deployment. The `ReleaseMap` dictionary is
+used to manage any overrides of repository and branch for a given component of the
+platform, as well as which components are included in that deployment.
+
+For example, to upgrade our MITx Residential deployments to start testing the Palm release we change the `CI` stages of the `mitx` and `mitx-staging` deployments to use the `palm` value for the `OpenEdxSupportedRelease`
+
+```diff
+@@ -13,7 +13,7 @@ class OpenLearningOpenEdxDeployment(Enum):
+     mitx = DeploymentEnvRelease(
+         deployment_name="mitx",
+         env_release_map=[
+-            EnvRelease("CI", OpenEdxSupportedRelease["olive"]),
++            EnvRelease("CI", OpenEdxSupportedRelease["palm"]),
+             EnvRelease("QA", OpenEdxSupportedRelease["olive"]),
+             EnvRelease("Production", OpenEdxSupportedRelease["olive"]),
+         ],
+@@ -21,7 +21,7 @@ class OpenLearningOpenEdxDeployment(Enum):
+     mitx_staging = DeploymentEnvRelease(
+         deployment_name="mitx-staging",
+         env_release_map=[
+-            EnvRelease("CI", OpenEdxSupportedRelease["olive"]),
++            EnvRelease("CI", OpenEdxSupportedRelease["palm"]),
+             EnvRelease("QA", OpenEdxSupportedRelease["olive"]),
+             EnvRelease("Production", OpenEdxSupportedRelease["olive"]),
+         ],
+```
+
+Because Palm is a new release for these deployments we also need to add a `palm` key to
+the `ReleaseMap` dictionary that contains the applications that are associated with
+those deployments and the appropriate `OpenEdxApplicationVersion` records for that
+deployment.
+
+```diff
+@@ -61,6 +61,122 @@ ReleaseMap: dict[
+     OpenEdxSupportedRelease,
+     dict[OpenEdxDeploymentName, list[OpenEdxApplicationVersion]],
+ ] = {
++    "palm": {
++        "mitx": [
++            OpenEdxApplicationVersion(
++                application="edx-platform",  # type: ignore
++                application_type="IDA",
++                release="palm",
++                branch_override="mitx/palm",
++                origin_override="https://github.com/mitodl/edx-platform",
++            ),
+             ...
++        ],
++        "mitx-staging": [
++            OpenEdxApplicationVersion(
++                application="edx-platform",  # type: ignore
++                application_type="IDA",
++                release="palm",
++                branch_override="mitx/palm",
++                origin_override="https://github.com/mitodl/edx-platform",
++            ),
+             ...
++        ],
++    },
+     "olive": {
+         "mitx": [
+             OpenEdxApplicationVersion(
+```
+
+All of the deployment pipelines for these application components are managed by a
```

Comment:
Where do all these moving parts live inside of Git? I think it would be useful to link to examples in the repo so we could see what the actual files look like in the context of their setting.